### PR TITLE
Recover the structured error report across the Temporal submitter boundary

### PR DESCRIPTION
**Pull request by lchoquel**
## Summary

Recovers the structured `ErrorReport` across the Temporal submitter boundary so distributed execution surfaces the same rich error as in-process runs.

- Add `ErrorReport.from_dict` — a strict inverse of `to_dict`, tolerant of version skew by dropping unknown keys.
- Reuse the error-report extraction helper on the submitter side instead of duplicating it; updated the existing `from_app_error` caller.
- Split `execute_workflow`'s combined `except` into a dedicated `except WorkflowFailureError` clause plus a combined `except (WorkflowAlreadyStartedError, RPCError)` clause.
- Recovery is guarded: a `ValidationError` from `from_dict` falls back to a generic `WorkflowExecutionError`.

## Test plan

- New unit tests for `from_dict` round-trip (nested `UserAction` / `ProviderErrorMetadata`, `provider_metadata` 429), workflow caller error recovery, child error recovery, and `WorkflowExecutionError`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Recover the structured `ErrorReport` across the Temporal submitter and child-workflow boundaries and propagate the real failure message, so distributed runs match local runs in CLI/HTTP classification and status mapping — including past wrapper `ApplicationError`s.

- **Bug Fixes**
  - Recover the report from `ApplicationError.details` on workflow failure; walk past wrapper `ApplicationError`s with empty details to the real payload.
  - Apply the same recovery for child-workflow failures in `execute_child_workflow` and `start_child_workflow`; keep generic behavior for `WorkflowAlreadyStartedError` and `RPCError`.
  - Guard recovery: drop unknown keys (version skew) and, on `ValidationError`, fall back to a generic `WorkflowExecutionError`.
  - Split `execute_workflow` handling with a dedicated `WorkflowFailureError` clause; propagate the original failure message.

- **New Features**
  - Added `ErrorReport.from_dict` (strict inverse of `to_dict`), exposed `error_report_dict_from_details`, and introduced `recover_error_report`.
  - `WorkflowExecutionError` now accepts an optional `error_report` and returns it from `to_error_report()` when present.
  - Added integration parity tests (local vs Temporal) with shared test data to lock classification and message parity end-to-end, plus unit tests for submitter and child-workflow recovery and edge cases.

<sup>Written for commit ac590d1e938fd088a0efafd568a64ea2aaa6eded. Summary will update on new commits. <a href="https://cubic.dev/pr/Pipelex/pipelex/pull/912?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: pipelex/temporal/tprl/temporal_error.py [hunk 1]
```diff
@@ -25,6 +27,55 @@ def _error_report_from_details(details: Sequence[Any]) -> dict[str, Any] | None:
...
+    ``ApplicationError``'s ``details``. The child-workflow boundary exposes its
+    failure via ``ChildWorkflowError.cause`` rather than ``__cause__``, so its
+    caller passes ``exc.cause`` straight in — already an ``ApplicationError``,
+    it is returned on the first iteration.
+    """
+    node: BaseException | None = exc
+    seen: set[int] = set()
+    while node is not None and id(node) not in seen:
+        if isinstance(node, ApplicationError):
+            return node
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/temporal/tprl/temporal_error.py]: ✅ Fixed — `_find_application_error` stopped at the first `ApplicationError` in the `__cause__` chain, so a report-less wrapper (a `WorkflowExecutionError` re-serialized when a workflow wraps a failed child workflow) hid the report-carrying child `ApplicationError` deeper down. Replaced it with `_find_error_report_dict`, which walks the whole chain and returns the first `ApplicationError` whose `details` actually carry a report. Added a regression test (`test_recovers_report_past_report_less_wrapper_application_error`).

File: wip/error-handling/track-temporal-integration.md [hunk 2]
```diff
@@ -36,13 +36,29 @@ The decorator catches `PipelexError` only (never the generic `Exception`) and re
...
 
 ## Related tracks
 
 - [track-retry-and-resilience.md](track-retry-and-resilience.md) — this is Tier 2; direct execution has no pipeline-level retry tier below it.
 - [track-worker-classification.md](track-worker-classification.md) — every inference worker classifies before this bridge is meaningful.
 - [track-metadata-model.md](track-metadata-model.md) — the `error_category` / `user_action` / `model` / `provider` on `to_error_report()` are what the details payload carries.
-- [track-cli-delivery.md](track-cli-delivery.md) — once `ApplicationError.details` carries the full report, the Temporal-side CLI / Web UI can render the same structured information as the agent JSON path.
+- [track-cli-delivery.md](track-cli-delivery.md) — the delivery surfaces consume `to_error_report()`; the submitter-side gap above is why that data source is degraded for a Temporal-run pipe.
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply wip/error-handling/track-temporal-integration.md]: ✅ Fixed — applied the suggested rewrite. The bullet now reads that, with submitter-side recovery landed, Temporal-run pipes preserve the same structured error data path as local runs — consistent with the doc's own landed sections.
